mono/metadata/Makefile.am

```diff
@@ -147,6 +147,39 @@ libmono_ilgen_la_LIBADD = $(glib_libs) ../utils/libmonoutils.la ../sgen/libmonos
 endif
 endif
 
+if !HOST_WIN32
+if !HOST_ANDROID
+if !HOST_IOS
```

Comment:
As we are trying to enable the FSEvent FileSystemWatcher on `2018-04` either we need to build with `HOST_IOS`, or the two tiny ported functions needed should stay in `metadata/filewatcher.*` for now with `#ifdef  TARGET_IOS`